modules/contracts/src.ts/services/ethService.ts

```diff
@@ -195,21 +197,38 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     txFn: (gasPrice: BigNumber, nonce: number) => Promise<TransactionResponse | undefined>,
     gasPrice: BigNumber,
     signer: Signer,
+    chainId: number,
     nonce?: number,
   ): Promise<Result<TransactionResponse | undefined, Error>> {
     // Queue up the execution of the transaction.
-    return await this.queue.add(
-      async (): Promise<Result<TransactionResponse | undefined, Error>> => {
-        try {
-          // Send transaction using the passed in callback.
-          const actualNonce: number = nonce ?? (await signer.getTransactionCount("pending"));
-          const response: TransactionResponse | undefined = await txFn(gasPrice, actualNonce);
+    if (!this.queues.has(chainId)) {
+      return Result.fail(new ChainError(ChainError.reasons.SignerNotFound));
+    }
+    // Define task to send tx with proper nonce
+    const task = async (): Promise<Result<TransactionResponse | undefined, Error>> => {
+      try {
+        // Send transaction using the passed in callback.
+        const stored = this.nonces.get(chainId);
+        const nonceToUse: number = nonce ?? stored ?? (await signer.getTransactionCount("pending"));
+        const response: TransactionResponse | undefined = await txFn(gasPrice, nonceToUse);
+        // After calling tx fn, set nonce to the greatest of
+        // stored, pending, or incremented
+        const pending = await signer.getTransactionCount("pending");
+        const incremented = nonceToUse + 1;
+        // Ensure the nonce you store is *always* the greatest of the values
+        const toCompare = stored ?? 0;
+        if (toCompare >= incremented && toCompare >= pending) {
+          // return without updating
```

Comment:
just to clean up - maybe we could reverse this if condition and put the `this.nonces.set` inside the if block

and then after the if block put the return statement